Main/Peripherals/src/Peripherals/CommandDispatcher.cpp

```diff
@@ -4,19 +4,24 @@
 
 #include "CommandDispatcher.h"
 
-CommandDispatcher::CommandDispatcher(std::istream& in, ThrustController& thrustController, PowerManager& powerManager, IHeadlights& lights)
-        : in_(in), thrustController_(thrustController), powerManager_(powerManager), lights_(lights), shouldExit_(false) {}
+CommandDispatcher::CommandDispatcher(ThrustController& thrustController, PowerManager& powerManager, IHeadlights& lights)
+        : thrustController_(thrustController), powerManager_(powerManager), lights_(lights), shouldExit_(false) {}
 
-void CommandDispatcher::runLoop() {
+void CommandDispatcher::runLoop(std::istream& in) {
     while(!shouldExit_) {
         std::string cmd;
-        std::getline(in_, cmd);
+        std::getline(in, cmd);
         std::stringstream ss(cmd);
-        dispatchCommand(ss);
+        if(!run(ss))
+            shouldExit_ = true;
     }
 }
 
-void CommandDispatcher::dispatchCommand(std::stringstream& cmdString) {
+int CommandDispatcher::run(std::stringstream& ss) {
+    return dispatchCommand(ss);
+}
+
+int CommandDispatcher::dispatchCommand(std::stringstream& cmdString) {
```

Comment:
I know this seems cleaner.  But as a general rule of thumb, functions that perform commands should not return values. And functions that perform queries should not change state. This is called command/query segregation and helps avoid side effects.
I don't think my saving an internal variable is the best solution either, though.